accumulator_updater/NOTES.md

```diff
@@ -0,0 +1,113 @@
+## Questions
+
+1.  Do we need to support multiple Whitelists?
+2.  Support multiple accumulators
+    1.  should each accumulator store a different type of data?
+        => implications for length of merkle proof
+    2.
+3.  authority?
+4.  how to know what went into the `AccumulatorAccount` (for deserializing/proofs)
+    1.  Header?
+
+## To Do
+
+1. map out flow for add/remove price & product (handled by xc_admin).
+2. map out flow for update price (pyth-agent)
+3. map out proposed flow for accumulator updater
+4. map out current e2e flow including client/target chain
+5. Need a way to map a combination of (PythAccountTypes, PythSchema) to the actual fields being used
+   1. e.g. (Price, Compact) => [price, expo, timestamp]
+6. Also need to be careful to preserve backwards compatibility
+   by only ever appending fields if the fields for a (PythAccountType, PythSchema) combination
+   ever change.
+   1. maybe safer to always lock the fields once they've been published and just add new schemas (e.g. a CompactV2)
+
+## Implemenation Notes/Questions:
+
+1. use anchor? solitaire? vanilla solana?
+   1. for anchor, if add/delete/updateAccount should support multiple accounts at once, use Option or old `&ctx.accounts.remaining_accounts` call?
+2. should AccumulatorMapping have its own consts
+
+## Accumulator Update Flow:
+
+### Add Price Account
+
+1.  pyth-agent
+2.  `pyth-contract` - add_price() ix with price account & accumulator-updater program
+3.  cpi call to accumulator_updater `add_account` ix
+
+### Update Price Account
+
+1. pyth-agent - update_price
+2. pyth-contract - update*price ix - always tries to aggregate `if clock.slot > latest_agg_price.pub_slot*`
+3. if pyth-contract::aggregate_price
+
+### Add Price "Compact" Account
+
+## Accumulator Schema (possibly pyth-schema instead)
+
+We could have multiple derivations of the PriceAccount that need to be included in the accumulator
+ex: `PriceAccount` (full) `PriceOnly` `PriceAndEma`
+
+### Problems
+
+1. all calls to add/delete/update\_<pyth_account> need to include these (from the pyth-agent)
+   There are 2 options for how to handle/implement this:
```

Comment:
I got confused to understand the problem. I think the problem that you like to discuss here is the upgradability of accounts. How we can upgrade and what is the process. One thing is that this we can also say that this change does not happen quite often. But my important question is that what changes in pyth-agent? It's like this my view:

We need to update the contract anyway. For pyth-agent when we change the accounts passed to it we need an upgrade; which means a new account or .... Is it anything other than this? If so, maybe another approach be account lookup tables.